hack/check-generate.sh

```diff
@@ -106,17 +106,33 @@ if which git &>/dev/null; then
     exit 1
   fi
 
-  echo ">> make tidy"
-  if ! out=$(make -f "$makefile" tidy 2>&1); then
-    echo "Error during calling make tidy: $out"
-    exit 1
-  fi
-  new_status="$(git status -s)"
+  repo_root="$(git rev-parse --show-toplevel)"
+  if [[ -d "$repo_root/vendor" ]]; then
+    echo ">> make revendor"
```

Comment:
I think you also should bring back the old boolean parameter if `make revendor` was called so that proper cleanup can be made: https://github.com/gardener/gardener/blob/8c02ffa7de2db1770dc5dc088b8e157622d29601/hack/check-generate.sh#L27
https://github.com/gardener/gardener/blob/8c02ffa7de2db1770dc5dc088b8e157622d29601/hack/check-generate.sh#L34
https://github.com/gardener/gardener/blob/8c02ffa7de2db1770dc5dc088b8e157622d29601/hack/check-generate.sh#L111

Reply:
I did the following change locally:
```diff
diff --git a/hack/check-generate.sh b/hack/check-generate.sh
index 6b2e1d0ff..a5569b009 100755
--- a/hack/check-generate.sh
+++ b/hack/check-generate.sh
@@ -23,14 +23,16 @@ check_branch="__check"
 initialized_git=false
 stashed=false
 checked_out=false
-generated=false
+executed_generate=false
+executed_revendor=false
+executed_tidy=false
 
 function delete-check-branch {
   git rev-parse --verify "$check_branch" &>/dev/null && git branch -q -D "$check_branch" || :
 }
 
 function cleanup {
-  if [[ "$generated" == true ]]; then
+  if [[ "$executed_generate" == true ]] || [[ "$executed_revendor" == true ]] || [[ "$executed_tidy" == true ]]; then
     if ! clean_err="$(make -f "$makefile" clean && git reset --hard -q && git clean -qdf)"; then
       echo "Could not clean: $clean_err"
     fi
@@ -93,7 +95,7 @@ if which git &>/dev/null; then
   fi
 
   echo ">> make generate"
-  generated=true
+  executed_generate=true
   if ! out=$(make -f "$makefile" generate 2>&1); then
     echo "Error during calling make generate: $out"
     exit 1
@@ -109,6 +111,7 @@ if which git &>/dev/null; then
   repo_root="$(git rev-parse --show-toplevel)"
   if [[ -d "$repo_root/vendor" ]]; then
     echo ">> make revendor"
+    executed_revendor=true
     if ! out=$(make -f "$makefile" revendor 2>&1); then
       echo "Error during calling make revendor: $out"
       exit 1
@@ -122,6 +125,7 @@ if which git &>/dev/null; then
     fi
   else
     echo ">> make tidy"
+    executed_tidy=true
     if ! out=$(make -f "$makefile" tidy 2>&1); then
       echo "Error during calling make tidy: $out"
       exit 1

```

After that I realised that it works as it is currently. There is no case where `executed_revendor`/`executed_tidy` would be true but `executed_generate` would be false.

Reply:
Ah right, there's no chance for an early exit in which the revendoring would run but `executed_generate` would be false.
Nice!